src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -883,6 +907,59 @@ IEnumerable<object> TryFindResource(ResourceDictionary resourceDictionary)
 				}
 			}
 		}
+
+		/// <summary>
+		/// Debugging helper method to get the parent element that this element is inheriting <paramref name="propertyName"/> from.
+		///
+		/// Eg, if you call textBlock.GetPropInheritanceParent("Foreground"), it will return the first parent which has a property called
+		/// "Foreground" (most likely the Control whose template contains the TextBlock).
+		/// </summary>
+		internal FrameworkElement GetPropInheritanceParent(string propertyName)
+		{
+			var dependencyProp = GetDPByName(propertyName);
+			if (!IsInherited(dependencyProp, GetType()))
+			{
+				return null;
```

Comment:
We should probably start moving those to a different partial class

Reply:
Makes sense, I'll move them.

Reply:
done